src/Microsoft.SqlTools.ServiceLayer/LanguageServices/LanguageService.cs

```diff
@@ -636,6 +650,101 @@ internal CompletionItem ResolveCompletionItem(CompletionItem completionItem)
+            // The suggestion title does not contain any sql punctuation
+            string tokenText = TextUtilities.RemoveSquareBracketSyntax(selectedToken.Text);
+
+            if (scriptParseInfo.IsConnected && Monitor.TryEnter(scriptParseInfo.BuildingMetadataLock))
+            {
+                try
+                {
+                    // Queue the task with the binding queue    
+                    QueueItem queueItem = this.BindingQueue.QueueBindingOperation(
+                        key: scriptParseInfo.ConnectionKey,
+                        bindingTimeout: LanguageService.PeekDefinitionTimeout,
+                        bindOperation: (bindingContext, cancelToken) =>
+                        {
+                            // Get suggestions for the token
+                            int parserLine = textDocumentPosition.Position.Line + 1;
+                            int parserColumn = textDocumentPosition.Position.Character + 1;
+                            IEnumerable<Declaration> declarationItems = Resolver.FindCompletions(
+                                scriptParseInfo.ParseResult, 
+                                parserLine, parserColumn, 
+                                bindingContext.MetadataDisplayInfoProvider);
+
+                            // Match token with the suggestions(declaration items) returned
+                            string schemaName = this.GetSchemaName(scriptParseInfo, textDocumentPosition.Position, scriptFile);
+                            PeekDefinition peekDefinition = new PeekDefinition(connInfo);
+                            return peekDefinition.GetScript(declarationItems, tokenText, schemaName);
+                            
+
+                        },
+                        timeoutOperation: (bindingContext) =>
+                        {
+                            return null;
+                        });
+
+                    // wait for the queue item
+                    queueItem.ItemProcessed.WaitOne();
+                    return queueItem.GetResultAsT<Location[]>();
+                }
+                finally
+                {
+                    Monitor.Exit(scriptParseInfo.BuildingMetadataLock);
+                }
+            }
+
+            return null;
+        }
+
+        /// <summary>
+        /// Extract schema name for a token, if present
+        /// </summary>
+        /// <param name="scriptParseInfo"></param>
+        /// <param name="position"></param>
+        /// <param name="scriptFile"></param>
+        /// <returns> schema nama</returns>
+        private string GetSchemaName(ScriptParseInfo scriptParseInfo, Position position, ScriptFile scriptFile)
+        {
+            // Get schema name 
+            string sql = scriptFile.GetLine(position.Line + 1);
+            int startColumn = TextUtilities.PositionOfPrevDelimeter(sql, 0, position.Character);
+            // if no schema name, returns null
+            if( sql[startColumn - 1] != '.')
```

Comment:
>sql[startColumn - 1] != '.') [](start = 16, length = 28)

We should get the last non-whitespace token instead of assuming the previous character works. The problem is that you can do "SELECT dbo. T1" with a space between the . and the table name. This is  valid and we have to cover these cases. Please also include a test or this scenario :)